### DISCUSSION TITLE
Code Block Content Tabs - Without Indent

### DISCUSSION THREAD

**Discussion by flywire** [squidfunk/mkdocs-material#3150] [Ideas]
https://squidfunk.github.io/mkdocs-material/reference/content-tabs/#grouping-code-blocks

> Code blocks are one of the primary targets to be grouped, and can be considered a special case of content tabs...

The blocks are already marked. Why not optionally support existing end of block as end of tab content instead of requiring indenting?

Should this be posted to https://github.com/facelessuser/pymdown-extensions for @facelessuser ?

**Comment by facelessuser** [comment]
Because it works better with how Python Markdown parses things. And yes, this is probably not a Material question.

**Reply by squidfunk** [reply]
This was practically the behavior of the legacy tabbed code blocks implementation, right?

**Reply by facelessuser** [reply]
I'm not sure what we are referring to here as "legacy".

If we are talking about the first tab implementation that was specific to only code, then it used a special variable in code headers:

````
```language tab="title"
...
```
````

This was built into SuperFences but **only** worked for code blocks.

Then we introduced the Tabbed extension which could allow any content of any complexity.

```
=== "Title"
    paragraph

    - list
    - list
    
    > block quote
```

The old way was retired and we only allow the generic way. This is not code block specific and doesn't care if you have one code block many code blocks or no code blocks.

Now if we are talking `Tabbed` vs `Tabbed` with `alternate_style = True`, nothing changed in syntax, just output.

**Reply by flywire** [reply]
I don't suppose I'm keen to run a depreciated version. Was there a conversion process for a file formatted for the first tab implementation? My use case is many many existing and new code snippets (eg [1](https://flywire.github.io/lo-p/47-Calc_Add-ins.html) or [2 [unformatted]](https://flywire.github.io/lo-p/15-Complex_Shapes.html)) I want to push into tabs. Then I want to replicate the tabs for other languages.

Why? It's convenient to add code without indenting it, maybe just an include statement (possibly recalling sphinx).

**Reply by facelessuser** [reply]
>  Was there a conversion process for a file formatted for the first tab implementation?

Just manually changing the syntax to match the new syntax. There is no automated process.

> Why? It's convenient to add code without indenting it, maybe just an include statement (possibly recalling sphinx).

I'm not sure exactly what is being asked here really. Unfortunately, I cannot change the syntax for a single user's use case. As I mentioned, part of the reason the syntax is implemented the way it is is due to Python Markdown's parser. And since it is not code block specific, I do not have a code block specific convenience approach.

**Reply by facelessuser** [reply]
It is possible that you could write your own conversion process.

**Reply by flywire** [reply]
Conversion using GNU sed on Windows:

```
sed '/^^```/{ s/^^/=== "Title"\n    /; :a; n; s/^^/    /; /^^    ```$/b; ba; }' file.md
```

[Source](https://stackoverflow.com/a/69748218/4539999)
